sdk/src/azure/core/az_credential_client_secret.c

```diff
@@ -110,7 +96,10 @@ AZ_NODISCARD az_result az_credential_client_secret_init_with_authority(
   _az_PRECONDITION_VALID_SPAN(tenant_id, 1, false);
   _az_PRECONDITION_VALID_SPAN(client_id, 1, false);
   _az_PRECONDITION_VALID_SPAN(client_secret, 1, false);
-  _az_PRECONDITION_VALID_SPAN(authority, 1, false);
+  _az_PRECONDITION_VALID_SPAN(authority, 0, true);
+
+  static az_span const azure_ad_global_authority
+      = AZ_SPAN_LITERAL_FROM_STR("https://login.microsoftonline.com/");
```

Comment:
Do we want to make this public?

Also, does this name make more sense?
```suggestion
  static az_span const default_azure_ad_global_authority
      = AZ_SPAN_LITERAL_FROM_STR("https://login.microsoftonline.com/");
```

Reply:
Not sure about this. Default *is* Azure AD global authority, Azure AD global authority is default. Azure AD global authority is the only one, there is no non-default Azure AD global authority. So I'd pick either `default_authority`, or `azure_ad_global_authority`, but not both. I think the way it is used, and from the comments is obvious it is a default value, so I'dpick the more descriptive name from the two.

Reply:
Terminology, see: https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud

Reply:
How about az_aad_global_authority.
I think we should prefix with az_ because it is in Azure Core.
aad is a well-known acronym for Azure AD
this is the global authority.
It is only a "default" in the context of when AZ_SPAN_NULL is passed when creating the credential. This string is not a default by itself; only in the context of something else.

Reply:
I agree with aad_global_authority. We don't necessarily have to preix it with az, because it is a local static, you can see it is defined in the function scope, it is not even visible in other places in this file, and not available to linker.

Reply:
> We don't necessarily have to preix it with az, because it is a local static, you can see it is defined in the function scope, it is not even visible in other places in this file, and not available to linker.

Sure, for now. And your suggestion is fine :)

But part of the discussion was whether we want to make it public and if so, what  would we call it.

Reply:
I actually don't have that strong of opinion here. So, `az_aad_global_authority` looks good?
I can declare it as static global, which still make it visible only inside this only `.c` file, and not visible to linker, but should we ever want to make it public/visible, we'll need to remove `static`, and add `extern` declaration to the header file.

Reply:
> add `extern` declaration to the header file.

Is the other option to move it into the `.h` file from the `.c` file rather than using `extern` feasible as well (if we wanted to do it in the future)?

Reply:
If we declare it in the header file, as extern in the future, an extra symbol will become visible to the user. Currently it is not.
We should not move *declarations* (az_span X = "ABC") to header files, because that results in the symbol being moved to customer variables and would be likely included multiple times to the binary.
So, we are good, we're compatible with doing the things in idiomatic way. For example, see how `az_context_application` is defined - it is `extern` in header and global in .c.